### PR TITLE
auth: Support all JWT algorithms

### LINKED ISSUE #9696: auth: Update JWT implementation to support other signature methods
The current JWT implementation uses the public key to verify a given JWT token from client, as shown in the following code in auth/jwt.go
31         verifyKey  *rsa.PublicKey
47         parsed, err := jwt.Parse(token, func(token *jwt.Token) (interface{}, error) {
 48                 return t.verifyKey, nil
 49         })

As explained in https://auth0.com/blog/critical-vulnerabilities-in-json-web-token-libraries/, this implementation is vulnerable to certain attacks that hackers can defeat JWT. To fix it, private key should be used to verify JWT tokens from client, instead of using the public key.

### PR DESCRIPTION
**Pull request by joelegasse**
This change adds support to etcd for all of the JWT algorithms included in the underlying JWT library.

Fixes #9696

### CODE REVIEW

File: auth/store.go [hunk 1]
```diff
@@ -66,6 +66,10 @@ var (
 	ErrInvalidAuthToken     = errors.New("auth: invalid auth token")
 	ErrInvalidAuthOpts      = errors.New("auth: invalid auth options")
 	ErrInvalidAuthMgmt      = errors.New("auth: invalid auth management")
+	ErrInvalidAuthMethod    = errors.New("auth: invalid auth signature method")
+	ErrMissingKey           = errors.New("auth: missing key data")
+	ErrKeyMismatch          = errors.New("auth: public and private keys don't match")
+	ErrVerifyOnly           = errors.New("auth: token signing attempted with verify-only key")
```
Comment (gyuho) [inline auth/store.go]: I don't see `ErrVerifyOnly` being used anywhere?

Comment (joelegasse) [reply auth/store.go]: Good catch, thanks.

Comment (joelegasse) [reply auth/store.go]: Updated, and added a test case to ensure it's returned.

**Comment by mitake** [comment]
lgtm, thanks! @joelegasse

**Pull request merged**